tests/ops/functions/test_equal.py

```diff
@@ -2726,124 +2723,97 @@ def v_function4(params):
+        new_ops = qml.tape.QuantumScript(other_op.hyperparameters["V"]).bind_new_parameters(
+            op_params, [0]
+        )
 
         new_other_op = deepcopy(other_op)
-        new_other_op.hyperparameters["v_function"] = op.hyperparameters["v_function"]
-        assert qml.equal(op, new_other_op) is True
+        new_other_op.hyperparameters["V"] = new_ops
+        assert qml.equal(op, new_other_op)
 
     @pytest.mark.parametrize("op, other_op", [(op1, op4)])
-    def test_non_equal_u_tapes(self, op, other_op):
-        """Test that differing u_tapes are found."""
-        assert qml.equal(op, other_op) is False
+    def test_non_equal_u_ops(self, op, other_op):
+        """Test that differing u operations are found."""
+        assert not qml.equal(op, other_op)
 
         new_other_op = deepcopy(other_op)
-        new_other_op.hyperparameters["u_tape"] = op.hyperparameters["u_tape"]
-        assert qml.equal(op, new_other_op) is True
-
-    @pytest.mark.parametrize("op, other_op", [(op1, op3_tapediff)])
-    def test_non_equal_v_tapes(self, op, other_op):
-        """Test that differing v_tapes are found."""
-        assert qml.equal(op, other_op) is False
-        # The v_function must have been different as well if it produced
-        # a different tape and the parameters and v_wires were fixed.
-        new_other_op = deepcopy(other_op)
-        new_other_op.hyperparameters["v_function"] = op.hyperparameters["v_function"]
-        new_other_op.hyperparameters["v_tape"] = op.hyperparameters["v_tape"]
-        assert qml.equal(op, new_other_op) is True
+        new_other_op.hyperparameters["U"] = op.hyperparameters["U"]
+        assert qml.equal(op, new_other_op)
 
     @pytest.mark.parametrize("op, other_op", [(op1, op7)])
     def test_non_equal_v_wires(self, op, other_op):
         """Test that differing v_wires are found."""
-        assert qml.equal(op, other_op) is False
-        # If the v_wires were different, so must have been the v_function and the
-        # resulting v_tape
+        assert not qml.equal(op, other_op)
         new_other_op = deepcopy(other_op)
-        new_other_op.hyperparameters["v_function"] = op.hyperparameters["v_function"]
-        new_other_op.hyperparameters["v_tape"] = op.hyperparameters["v_tape"]
-        new_other_op.hyperparameters["v_wires"] = op.hyperparameters["v_wires"]
-        assert qml.equal(op, new_other_op) is True
+        new_other_op.hyperparameters["V"] = op.hyperparameters["V"]
+        assert qml.equal(op, new_other_op)
 
     @pytest.mark.parametrize("op, other_op", [(op5, op6), (op1_trainable, op1_untrainable)])
     def test_trainability(self, op, other_op):
         """Test that differing trainabilities are found."""
-        assert qml.equal(op, other_op) is False
-        assert qml.equal(op, other_op, check_trainability=False) is True
+        assert not qml.equal(op, other_op)
+        assert qml.equal(op, other_op, check_trainability=False)
```

Comment:
Mind returning them to 
```suggestion
        assert qml.equal(op, other_op) is False
        assert qml.equal(op, other_op, check_trainability=False) is True
```
here and in other places? We are wanting to actually test that the outputs are `False` and `True`, not just truthy and falsy.  For example, we don't want this test to "pass" if `qml.equal` is returning a string or some other weird thing.

Reply:
This is very smart observation.
I replaced the equal tests for the HilbertSchimdt case and a few others in the file. There are 300+ entries like this in this file and right now I don't have the mental capacity to replace all of them, but this is something I didn't think about. Thanks!